src/Pages/AuditLog/Components/Pagination.js

```diff
@@ -0,0 +1,66 @@
+const Pagination = ({ currentPage, totalPages, goToPage, startIndex, endIndex }) => {
+  return (
+    <div className='mt-8 flex items-center justify-between'>
+      <div className='text-sm text-gray-400'>
+        Showing {startIndex + 1} to {endIndex} of {endIndex} results
+      </div>
+
+      <div className='flex items-center space-x-2'>
+        <button
+          onClick={() => goToPage(currentPage - 1)}
+          disabled={currentPage === 1}
+          className={`px-3 py-2 rounded-md text-sm font-medium ${
+            currentPage === 1
+              ? 'bg-gray-700 text-gray-500 cursor-not-allowed'
+              : 'bg-gray-800 text-gray-300 hover:bg-gray-700 border border-gray-600'
+          }`}
+        >
+          Previous
+        </button>
+
+        <div className='flex items-center space-x-1'>
+          {[...Array(Math.min(5, totalPages))].map((_, i) => {
+            let pageNum;
+            if (totalPages <= 5) {
+              pageNum = i + 1;
+            } else if (currentPage <= 3) {
+              pageNum = i + 1;
+            } else if (currentPage >= totalPages - 2) {
+              pageNum = totalPages - 4 + i;
+            } else {
+              pageNum = currentPage - 2 + i;
+            }
```

Comment:
could a comment be added above each assignment of pageNum to explain what it means? it wouldnt be clear to those unfamiliar with the code (including me)

also you can remove the else
```suggestion
            let pageNum = currentPage - 2 + i;
            if (totalPages <= 5) {
              pageNum = i + 1;
            } else if (currentPage <= 3) {
              pageNum = i + 1;
            } else if (currentPage >= totalPages - 2) {
              pageNum = totalPages - 4 + i;
            }
```